garrysmod/lua/includes/modules/worldtip.lua

```diff
@@ -0,0 +1,56 @@
+
+local IsValid = IsValid
+local SysTime = SysTime
+local isvector = isvector
+
+
+/*
+	Name: worldtip
+	Desc: Handles world/entity-based tooltips
+*/
+module( "worldtip" )
+
+local WorldTips = {}
+
+function GetTable()
+	return WorldTips
+end
+
+function Add( name, tgt, text, len, col )
+
+	local ent, pos = IsValid( tgt ) && tgt
+
+	if ent then
+		if ent:IsWorld() then return end
+		ent:CallOnRemove( "RemoveEntityTooltip", function() WorldTips[name] = nil end )
+	else
+		if !isvector(pos) then return end
```

Comment:
Yeah, I've been too busy to look at this recently. I'll dust it off later if I get an hour.